aspnetcore/security/authentication/index.md

```diff
@@ -0,0 +1,107 @@
+---
+# Overview of ASP.NET Core authentication
+
+By [Mike Rousos](https://github.com/mjrousos)
+
+Authentication is the process of determining a user's identity. [Authorization](xref:security/authorization/introduction) is the process of determining whether a user has access to a resource. In ASP.NET Core, authentication is handled by authentication [middleware](xref:fundamentals/middleware/index). The authentication middleware uses registered authentication handlers to complete authentication-related actions. Examples of authentication-related actions include:
+
+* Authenticating a user.
+* Responding when an unauthenticated user tries to access a restricted resource.
+
+The registered authentication handlers and their configuration options, are called "schemes".
+
+Authentication schemes are specified by registering authentication services in `Startup.ConfigureServices`. This is done by:
+
+* Calling [AuthenticationBuilder.AddScheme](xref:Microsoft.AspNetCore.Authentication.AuthenticationBuilder.AddScheme*)
+* Calling a scheme-specific extension methods that call `AddScheme` automatically with appropriate settings. This approach is more common than calling `AuthenticationBuilder.AddScheme`.
+
+For example, the following code registers authentication services and handlers for cookie and JWT bearer authentication schemes:
+
+```csharp
+services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+    .AddJwtBearer(options => Configuration.Bind("JwtSettings", options))
+    .AddCookie(options => Configuration.Bind("CookieSettings", options));
+```
+
+The `AddAuthentication` parameter `JwtBearerDefaults.AuthenticationScheme` is the name of the scheme to use by default if a specific one isn't requested. In the preceding code, JWT bearer authentication is used by default.
+
+In some cases, the call to `AddAuthentication` is automatically made by other extension methods. For example, when using [ASP.NET Core Identity](xref:security/authentication/identity)),`AddAuthentication` is called. If multiple schemes are used, authorization policies (or authorization attributes) can [specify the authentication scheme (or schemes)](xref:security/authorization/limitingidentitybyscheme) they depend on to authenticate the user.
+
+Authentication middleware is added in `Startup.Configure` by calling the <xref:Microsoft.AspNetCore.Builder.AuthAppBuilderExtensions.UseAuthentication*> extension method on the app's `IApplicationBuilder`. This registers the  middleware which uses the previously registered authentication schemes. Call `UseAuthentication` before any middleware that depends on users being authenticated. When using endpoint routing, the call to `UseAuthentication` should go:
+
+* After `UseRouting`, so that route information is available for authentication decisions.
+* Before `UseEndpoints`, so that users are authenticated before accessing the endpoints.
+
+## Authentication Concepts
+
+### Authentication scheme
+
+An authentication scheme is:
```

Comment:
@HaoK how do you feel about this? To me a scheme is just a dictionary key for an authentication handler. I'm not sure I'd say the scheme is the way of authenticating the user or responding if there's an unauthenticated or forbidden user, that's the handler.

Reply:
The bullets here are just referring to the different methods the authentication handler supplies (Authenticate, Challenge, Forbid). As you say, a scheme is just a key for a handler (along with configuration, since the same handler might be registered multiple times as different schemes) but I want to define scheme, specifically, since it's used fairly often (and authentication handler is not) and I know some customers have been confused about what the term means.

Maybe the better approach is to define an authentication scheme as the *name* of an authentication mechanism? Something like: "An authentication scheme is a name which corresponds to an authentication handler and options for configuring that specific instance of the handler. Schemes are useful as a mechanism for easily referring to the authentication, challenge, and forbid behaviors of the associated handler."

Thoughts?

Reply:
Yes that's exactly how I would describe it:

Authentication Scheme = the logical name of one particular instance of an authentication handler.  This also helps when discussing virtual schemes and how that opens up redirection specific authentication actions to different handlers/other schemes.